### PR TITLE
Fixup `index_setsm.py` dryrun actions

### PR DESCRIPTION
**Pull request by ehusby**
Currently `index_setsm.py` will create the output database index file/layer when the script is run in `--dryrun` mode. I don't think this is desired behavior?

These changes would limit the scope of a dryrun to just the "Identifying DEMs" step. Example output:
```
04-15-2022 12:40:27 INFO- Source: /home/husby036/scratch/_temp/WV02_20140311_103001002DB55100_103001002EA1A600_2m_lsf_v030301/
04-15-2022 12:40:27 INFO- Identifying DEMs
[============================================================] 100.0% ...DEMs identified
04-15-2022 12:40:27 INFO- 1 records found
04-15-2022 12:40:27 INFO- Exiting dryrun
```

### CODE REVIEW

**Comment by clairecporter** [comment]
It's a bug and a feature, actually.  The --dryrun option is sometimes used to initialize a new table with no records in postgres, for example when making a srtip_dem_all table.  If you want to repair this bug, we should have a new option that initializes a new table or revisit the workflow to make sure that functionality is no longer needed.

**Review by clairecporter — APPROVED** [review]
This looks fine.  I will retract my requirement to have the ability to initialize an empty table.  Did you test it back when you wrote it?

**Comment by ehusby** [comment]
I'm not sure how I feel about the changes in this PR as a whole. It's kinda two "fixes", which I'll highlight in comments.
- One change is simple, never allowing going into `write_to_ogr_dataset` if you're in dryrun mode
- The second change is fixing a small undefined variable error bug in `write_to_ogr_dataset` if the `ogr_driver_str` output format is not handled.

`write_to_ogr_dataset` still has a lot of logic left in it concerning dryrun mode, alluding to how you've used it to initialize a new table.

Even if we don't want to have it initialize a new table in dryrun mode, maybe it's still useful to let the logic in `write_to_ogr_dataset` run if you can get important info prints out of it.

File: index_setsm.py [hunk 1]
```diff
@@ -420,17 +420,22 @@ def main():
         #### Write index
         if args.write_json:
             write_to_json(dst, groups, total, args)
-        else:
+        elif not args.dryrun:
             write_result = write_to_ogr_dataset(ogr_driver_str, ogrDriver, dst_ds, dst_lyr, groups,
```
Comment (ehusby) [inline index_setsm.py:424]: With my changes here, `write_to_ogr_dataset` is just never entered in dryrun mode.

**Comment by clairecporter** [comment]
I resolved the merge conflicts so this is good to go

**Pull request merged**